### PR TITLE
adding few more c++17 operators override.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: src/snmalloc/override/new.cc [hunk 1]
```diff
@@ -71,3 +71,45 @@ void operator delete[](void* p, std::nothrow_t&)
 {
   ThreadAlloc::get().dealloc(p);
 }
+
+void* operator new(size_t size, std::align_val_t)
```
Comment (nwf-msr) [inline src/snmalloc/override/new.cc:75]: Discarding the alignment requirement seems dubious?  While it's generally true that `snmalloc` prefer to align things much more strongly than generally required, it's still possible to ask for yet stronger alignment.  I think the correct thing to do is to use https://github.com/microsoft/snmalloc/blob/d0df2d028a45131e5b85dce873c0bcc572f6a57d/src/mem/sizeclasstable.h#L131 as done with https://github.com/microsoft/snmalloc/blob/d0df2d028a45131e5b85dce873c0bcc572f6a57d/src/override/malloc.cc#L158-L174
I do not know to what extent those checks are also required by C++ .

File: src/snmalloc/override/new.cc [hunk 2]
```diff
@@ -71,3 +71,45 @@ void operator delete[](void* p, std::nothrow_t&)
...
+{
+  return ThreadAlloc::get().alloc(size);
+}
+
+void* operator new[](size_t size, std::align_val_t, std::nothrow_t&)
+{
+  return ThreadAlloc::get().alloc(size);
+}
+
+void operator delete(void* p, std::align_val_t)EXCEPTSPEC
```
Comment (nwf-msr) [inline src/snmalloc/override/new.cc:95]:
```suggestion
void operator delete(void* p, std::align_val_t) EXCEPTSPEC
```

I'm surprised `clang-format` was OK with that without the whitespace.

File: src/snmalloc/override/new.cc [hunk 3]
```diff
@@ -71,3 +71,45 @@ void operator delete[](void* p, std::nothrow_t&)
...
+{
+  ThreadAlloc::get().dealloc(p);
+}
+
+void operator delete[](void* p, std::align_val_t) EXCEPTSPEC
+{
+  ThreadAlloc::get().dealloc(p);
+}
+
+void operator delete(void* p, size_t size, std::align_val_t)EXCEPTSPEC
```
Comment (nwf-msr) [inline src/snmalloc/override/new.cc:105]:
```suggestion
void operator delete(void* p, size_t size, std::align_val_t) EXCEPTSPEC
```

**Review by nwf-msr — DISMISSED** [review]

File: src/snmalloc/override/new.cc (hunk 2 above)
Comment (nwf-msr) [reply src/snmalloc/override/new.cc]: ... oh wow, it in fact requires that it not be there.  That... that sure seems like a bug, but I guess it's best to do what it wants.  Sorry!

Comment (mjp41) [reply src/snmalloc/override/new.cc]: @nwf-msr sorry I am confused now.  I think @devnexen has now added the space, but are you saying the space shouldn't be there?

**Comment by mjp41** [comment]
@devnexen would be able to run clang-format to unblock the CI?

File: src/snmalloc/override/new.cc (hunk 2 above)
Comment (devnexen) [reply src/snmalloc/override/new.cc:99]: by the way is clang-format9 still recommended ? Thankfully I still have it somewhere.

Comment (mjp41) [reply src/snmalloc/override/new.cc:99]: It is, but I might move this and Verona forward to a newer version.  It is getting a bit long in the tooth now.

**Review by mjp41 — APPROVED** [review]

**Pull request merged**